synapse/handlers/sync.py

```diff
@@ -1819,8 +1819,12 @@ async def _generate_sync_entry_for_rooms(
 
         # Retrieve rooms that got un partial stated in the meantime, only useful in case
         # of a non lazy-loading-members sync.
+        # We also skip calculating that in case of initial sync since we don't need it.
```

Comment:
I'm not following you here. I thought we want to exclude partially-joined rooms from all non-lazy load syncs. Why should that not also include an initial sync?

Reply:
We skip calculating the rooms that got unpartialstated because in case of initial, we have [another piece of code](https://github.com/matrix-org/synapse/pull/14831/files#diff-70c84622edff898486bdc02d7a579db4e52a3e0b26c14020bed0ca1fb8ea2018R2239) that already exclude any partial stated rooms. The stream of unpartialstated rooms is only useful to know which ones have been unpartialed between 2 syncs, for an initial sync we just directly check the DB.